src/herder/test/HerderTests.cpp

```diff
@@ -362,10 +366,11 @@ testTxSet(uint32 protocolVersion)
             }
             SECTION("gap middle")
             {
-                int remIdx = 2; // 3rd transaction
+                int remIdx = protocolVersion == 10 ? 6 : 8;
```

Comment:
What's going on here?

Reply:
This test starts off with 10 total transactions from two accounts. It then removes the first transaction from one of the accounts where the transaction is in the middle of the set. 

The ordering of the transactions just happened to work out in a way where the first tx was in the same index between protocol versions (due to v0 and v1 envelopes), and that was no longer true after my change, so I just used specific versions. After thinking about this though, I'm not 100% what's going on here. Does `sortForApply` update the ordering of `mTransactions` by hash? I think the answer is yes but it's not immediately clear to me where that happens.

Reply:
The test is wrong - instead of going back and forth here, I added a commit to https://github.com/stellar/stellar-core/pull/3373 to fix it